### PR TITLE
Pie: Fix label display and update

### LINKED ISSUE #989: pie chart: labels are not displayed on creation
When I create a Pie chart using bqplot in a jupyter notebook, labels are not displayed.

**To Reproduce**
Here is the code I use:
```python
import numpy as np
import bqplot

pie = bqplot.marks.Pie(
                sizes=np.array([1, 2, 3]),
                labels=['foo', 'bar', 'baz']
            )
bqplot.Figure(marks=[pie], title="simple pie")
```
**Expected behavior**
I expect the labels ("foo", "bar", "baz") to appear on the chart.

**Screenshots**
image

**Environment (please complete the following information):**
- Operating System and Version: Debian Linux 10.2 "buster" 64bits
- Browser [e.g. Firefox, Safari] (if applicable): Firefox Quantum 68.2.0esr
- Python version 3.7.3
- Virtual env:
```bash
pip install -r requirements.txt
jupyter labextension install @jupyter-widgets/jupyterlab-manager --no-build
jupyter labextension install bqplot --no-build
jupyter labextension install jupyter-leaflet --no-build
jupyter labextension install @jupyter-voila/jupyterlab-preview --no-build
jupyter lab build
```
- requirements.txt
```
bqplot==0.12.1
ipyleaflet==0.11.6
ipywidgets==7.5.1
jupyterlab==1.2.3
palettable==3.3.0
pandas==0.25.3
requests==2.22.0
voila==0.1.20
```

**Additional context**
This problem occurs on `Jupyter Lab` and also on `Voila`

If this can help, I noticed that if I changed the labels of the graph after it was created, labels are correctly displayed on the chart
```python
pie.labels = ['Huey', 'Dewey', 'Louie']
``` 
image

Is it the way it's supposed to work?

### PR DESCRIPTION
**Pull request by martinRenou**
Signed-off-by: martinRenou <martin.renou@gmail.com>

Fix #989

### CODE REVIEW

**Comment by martinRenou** [comment]
This needs testing

**Comment by martinRenou** [comment]
Done

**Comment by pierredittgen** [comment]
Hi @martinRenou,

Thanks for this PR, I would like to review it. How can I test it in my dev environment?
- I (pip) uninstall bqplot module
- I cloned your [repo](https://github.com/martinRenou/bqplot),
- checkout `fix_pie_labels` branch
- install (pip install -e .) your python module

But when I try to enable extension, i got an error:
```bash
jupyter nbextension enable --py bqplot
Enabling notebook extension bqplot/extension...
      - Validating: problems found:
        - require?  X bqplot/extension
```

Any clue?

**Comment by martinRenou** [comment]
Unfortunately, my PR does not work currently. I need to practice more on the d3 `enter`, `exit` features which I am really not comfortable with.

If you want to try it anyway, I suspect you forgot to install the nbextension with:
```
jupyter nbextension install --py --symlink --sys-prefix bqplot
```
Did you follow exactly the development installation steps? https://github.com/bloomberg/bqplot#installation

**Comment by martinRenou** [comment]
This should be ready for a review/merge

File: js/src/Pie.ts [hunk 1]
```diff
@@ -250,62 +248,67 @@ export class Pie extends Mark {
...
-        labels.enter()
+        const labelsTransition = labels.enter()
             .append("text")
             .attr("dy", ".35em")
-            .style("opacity", 0)
-            .text(function(d) {
-                return d.data.label;
-            })
+            .text(d => d.data.label)
+            .style("opacity", 1)
```
Comment (SylvainCorlay) [inline js/src/Pie.ts]: 0?

**Pull request merged**